File: app/src/main/java/com/veyndan/paper/reddit/api/xkcd/network/XkcdService.java
package com.veyndan.paper.reddit.api.xkcd.network;

import android.support.annotation.NonNull;

import com.veyndan.paper.reddit.api.xkcd.model.XkcdComic;

import io.reactivex.Single;
import retrofit2.Response;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface XkcdService {

    @NonNull
    @GET("{num}/info.0.json")
    Single<Response<XkcdComic>> num(
            @Path("num") int num);
}

File: app/src/main/java/com/veyndan/paper/reddit/api/xkcd/network/XkcdService.kt
package com.veyndan.paper.reddit.api.xkcd.network

import com.veyndan.paper.reddit.api.xkcd.model.XkcdComic
import io.reactivex.Single
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Path

interface XkcdService {

    @GET("{num}/info.0.json")
    fun num(@Path("num") num: Int): Single<Response<XkcdComic>>
}
